static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+import {ArrowProps} from 'sentry/icons/iconArrow';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [anchorRefs, setAnchorRefs] = useState<HTMLElement[]>([]);
+  const [childrenRefs, setChildrenRefs] = useState<HTMLElement[]>([]);
+  const [isAtStart, setIsAtStart] = useState(true);
+  const [isAtEnd, setIsAtEnd] = useState(false);
+
+  useEffect(() => {
+    if (!ref.current) {
+      return () => {};
+    }
+
+    const observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(e => {
+          if (e.target.id === anchorRefs[0].id) {
+            setIsAtStart(e.isIntersecting);
+          } else if (e.target.id === anchorRefs[1].id) {
+            setIsAtEnd(e.isIntersecting);
+          }
+        });
+      },
+      {root: ref.current, threshold: [1]}
+    );
+
+    if (anchorRefs) {
+      anchorRefs.map(anchor => observer.observe(anchor));
+    }
+
+    return () => observer.disconnect();
+  }, [anchorRefs]);
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    setChildrenRefs(Array.from(ref.current.children) as HTMLElement[]);
+    const anchors = [
+      ref.current.children[0],
+      ref.current.children[ref.current.children.length - 1],
+    ] as HTMLElement[];
+    setAnchorRefs(anchors);
+  }, [children]);
+
+  const handleScroll = (direction: string) => {
+    requestAnimationFrame(() => {
```

Comment:
Do we still need to call `requestAnimationFrame()` now that we're using `scrollTo()`?

Reply:
actually, we do not!